### PR TITLE
feat(util): file downloader with verify sha256 hash

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

File downloader with check hash after download

### CODE REVIEW

File: util/downloader/downloader_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		if r.URL.Path == fileURL {
+			_, err := w.Write(fileContent)
+			assert.NoError(t, err)
+		} else {
+			http.NotFound(w, r)
+		}
+	}))
+	defer server.Close()
+
+	filePath := "./testdata/example_testfile.txt"
```
Comment (b00f) [inline util/downloader/downloader_test.go:33]: Use `util.TempFileName()` instead

**Review by b00f — APPROVED** [review]
Everything look good. One comment: We can use temp file for tests.

**Review by kehiy — APPROVED** [review]
Everything looks good to me.

File: .gitignore [hunk 2]
```diff
@@ -14,3 +14,5 @@ todo
 # VIM artifacts
 .swp
 .*.sw*
+
+util/downloader/testdata/
```
Comment (b00f) [inline .gitignore:18]: We can revert this line?

**Review by b00f — APPROVED** [review]

**Pull request merged**